plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerTypeMapping.java

```diff
@@ -69,6 +72,107 @@ public void testVarbinary()
+
+                .execute(getQueryRunner(), trinoCreateAndInsert(getSession(), "test_time"));
+
+        SqlDataTypeTest.create()
+                // round down
+                .addRoundTrip("TIME '00:00:00.00000001'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.000000000001'", "TIME '00:00:00.0000000'")
+
+                // round down, maximal value
+                .addRoundTrip("TIME '00:00:00.00000004'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.000000049'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.0000000449'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.00000004449'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.000000044449'", "TIME '00:00:00.0000000'")
+
+                // round up to next day, minimal value
+                .addRoundTrip("TIME '23:59:59.99999995'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '23:59:59.999999950'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '23:59:59.9999999500'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '23:59:59.99999995000'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '23:59:59.999999950000'", "TIME '00:00:00.0000000'")
+
+                // round up to next day, maximal value
+                .addRoundTrip("TIME '23:59:59.99999999'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '23:59:59.999999999'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '23:59:59.9999999999'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '23:59:59.99999999999'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '23:59:59.999999999999'", "TIME '00:00:00.0000000'")
+
+                // round down
+                .addRoundTrip("TIME '23:59:59.999999949999'", "TIME '23:59:59.9999999'")
+
+                .execute(getQueryRunner(), trinoCreateAndInsert(getSession(), "test_time"));
+
+        SqlDataTypeTest.create()
+                // round down
+                .addRoundTrip("TIME '00:00:00.00000001'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.000000000001'", "TIME '00:00:00.0000000'")
+
+                // round down, maximal value
+                .addRoundTrip("TIME '00:00:00.00000004'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.000000049'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.0000000449'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.00000004449'", "TIME '00:00:00.0000000'")
+                .addRoundTrip("TIME '00:00:00.000000044449'", "TIME '00:00:00.0000000'")
+
+                // round up to next day, minimal value
+                // SQL Server truncates instead of rounding up and recognizes only up to 9 digits
```

Comment:
Reading this comment back makes me think that the connector's `toWriteMapping` should throw an error if a literal would have precision greater than 9.

Reply:
I think it should round to precision 9 if the value has precision > 9.
Let's do what Postgres connector does.